state/unit.go

```diff
@@ -2162,6 +2143,9 @@ func assignContextf(err *error, unitName string, target string) {
 // AssignToMachine assigns this unit to a given machine.
 func (u *Unit) AssignToMachine(m *Machine) (err error) {
 	defer assignContextf(&err, u.Name(), fmt.Sprintf("machine %s", m))
+	if u.doc.Principal != "" {
+		return fmt.Errorf("unit is a subordinate")
```

Comment:
errors.NotSupportedf()

Reply:
This is out of scope for the current work as it changes the structure of the error message. We should make a separate pass through the unit code to update errors. There are many non-current things in here.